example_flutter/lib/main.dart

```diff
@@ -27,46 +30,108 @@ class MyApp extends StatefulWidget {
 }
 
 class _AppState extends State<MyApp> {
+  Color _primaryColor = Colors.blue;
+  int _counter = 0;
+
   static _AppState of(BuildContext context) =>
       context.ancestorStateOfType(const TypeMatcher<_AppState>());
-  Color _primaryColor = Colors.blue;
 
+  /// Sets the primary color of the example app.
   void setPrimaryColor(Color color) {
     setState(() {
       _primaryColor = color;
     });
   }
 
+  void incrementCounter() {
+    _setCounter(_counter + 1);
+  }
+
+  void _decrementCounter() {
+    _setCounter(_counter - 1);
+  }
+
+  void _setCounter(int value) {
+    setState(() {
+      _counter = value;
+    });
+  }
+
+  /// Rebuilds the native menu bar based on the current state.
+  void updateMenubar() {
+    // Currently the menubar plugin is only implemented on macOS.
+    if (!Platform.isMacOS) {
+      return;
+    }
+    setApplicationMenu([
```

Comment:
Since nothing else in the app depends on the menu update having happened, proceeding in parallel is fine.

However I'm really glad you asked, as it made me realize I didn't connect the async/await all the way through to the platform channel message, so setApplicationMenu would always return immediately instead of when the native side responded, making it impossible for a client who did want to know when the update completed to find out. That's been fixed now (even though this call site still doesn't wait).